assert/assertions_test.go

```diff
@@ -2986,6 +2986,28 @@ Diff:
 
 	expected = `
 
+Diff:
+--- Expected
++++ Actual
+@@ -12,3 +12,3 @@
+   x: (int) 4
+- }: (int) 4
++ }: (int) 999
+ }
+`
+
+	type Key struct {
+		x int
+	}
+
+	actual = diff(
+		map[Key]int{Key{1}: 1, Key{2}: 2, Key{3}: 3, Key{4}: 4},
+		map[Key]int{Key{1}: 1, Key{2}: 2, Key{3}: 3, Key{4}: 999},
+	)
+	Equal(t, expected, actual)
```

Comment:
I would like you to change a few things

- The value should differ from the key in the map. Otherwise it's unclear how it's ordered.

- your example should also display wgat happen when a value is added and removed

- there is no need for a local exported type

Please consider something like this

```suggestion
	type key struct {
		x int
	}

	actual = diff(
		map[key]int{key{1}: 789, key{2}: 678, key{3}: 567, key{4}: 456},
		map[key]int{key{2}: 678, key{3}: -1, key{4}: 456, key{5}: 345},
	)
	Equal(t, expected, actual)
```

Reply:
@ccoVeille: I think that's a different kind of test case that is trying to assert a *particular* order for the map, while I would argue that we only need a *deterministic* order. The fact that the current non-deterministic order produces unreadable diffs should be considered a bug (otherwise `SortKeys` wouldn't be enabled in the first place).

So while I could remove a key, add a key and replace a value in this test case, the problem is that it makes the diff much more confusing:

```diff
--- Expected
+++ Actual
@@ -1,5 +1,2 @@
 (map[assert.key]int) (len=4) {
- (assert.key) {
-  x: (int) 1
- }: (int) 789,
  (assert.key) {
@@ -9,6 +6,9 @@
   x: (int) 3
- }: (int) 567,
+ }: (int) -1,
  (assert.key) {
   x: (int) 4
- }: (int) 456
+ }: (int) 456,
+ (assert.key) {
+  x: (int) 5
+ }: (int) 345
 }
```

It's no longer obvious what we're testing and the diff is about as unreadable as the original broken diff. If someone were to break this test case in the future, there's a good chance they might just decide to replace the diff with whatever it spat out or delete the test case entirely.

I think my test case is about as simple as it can be while asserting that the map order is deterministic. It would be obvious that something has gone wrong if someone were to break this in the future because the diff would become unreadable again.